projects/Mox/module.nix

```
@@ -0,0 +1,84 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  options = {
+    services.mox = {
+      enable = lib.mkEnableOption "Enable Mox Mail Server";
+      configFile = lib.mkOption {
+        type = lib.types.str;
+        default = "/var/lib/mox/config/mox.conf";
+        description = "Path to the Mox configuration file";
+      };
+      hostname = lib.mkOption {
+        type = lib.types.str;
+        default = "mail";
+        description = "Hostname for the Mox Mail Server";
+      };
+      user = lib.mkOption {
+        type = lib.types.str;
+        description = "*Required* Email user as (user@domain) to be created.";
+      };
+    };
+  };
+
+  config = lib.mkIf config.services.mox.enable {
+    # Ensure neccesary packages are installed
+    environment.systemPackages = with pkgs; [
+      mox
+    ];
+
+    # Create a dedicated user/group for Mox
+    users.users.mox = {
+      isSystemUser = true;
+      name = "mox";
+      group = "mox";
+      home = "/var/lib/mox";
+      createHome = true;
+      description = "Mox Mail Server User";
+    };
+    users.groups.mox = {
+      name = "mox";
+    };
+
+    systemd.services.mox-setup = {
+      description = "Setup Mox Mail Server Configuration Files";
+      wantedBy = [ "multi-user.target" ];
+      requires = [
+        "network-online.target"
+        "unbound.service"
+      ];
+      after = [
+        "network-online.target"
+        "unbound.service"
+      ];
+      before = [ "mox.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        User = "mox";
+        Group = "mox";
```

We can just use [stateDirectory](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RuntimeDirectory=) (which is relative to `/var/lib`) and [WorkingDirectory](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#WorkingDirectory=)

```suggestion
        Group = "mox";
        StateDirectory = "mox";
        WorkingDirectory = "/var/lib/mox";
```